File: benchmark/python/strings.py
#!/usr/bin/env python3
# strings.py - String Operations Benchmark for Python
# Tests string concatenation and manipulation performance

import time

def count_occurrences(s, substr):
    """Count occurrences of substring"""
    count = 0
    pos = 0
    while True:
        pos = s.find(substr, pos)
        if pos == -1:
            break
        count += 1
        pos += 1
    return count

def main():
    count = 100000
    hello = "Hello"

    start = time.time()

    # Build string using list join (efficient in Python)
    parts = [hello] * count
    result = "".join(parts)

    # Count occurrences of 'llo'
    occurrences = count_occurrences(result, "llo")

    elapsed = int((time.time() - start) * 1000)

    print(f"String length: {len(result)}")
    print(f"Occurrences of 'llo': {occurrences}")
    print(f"String time: {elapsed}ms")

if __name__ == "__main__":
    main()

File: benchmark/nodejs/strings.js
#!/usr/bin/env node
// strings.js - String Operations Benchmark for Node.js
// Tests string concatenation and manipulation performance

function countOccurrences(str, substr) {
    let count = 0;
    let pos = 0;
    while ((pos = str.indexOf(substr, pos)) !== -1) {
        count++;
        pos++;
    }
    return count;
}

function main() {
    const count = 100000;
    const hello = "Hello";

    const start = Date.now();

    // Build string using array join (efficient in JS)
    const parts = new Array(count).fill(hello);
    const result = parts.join("");

    // Count occurrences of 'llo'
    const occurrences = countOccurrences(result, "llo");

    const elapsed = Date.now() - start;

    console.log(`String length: ${result.length}`);
    console.log(`Occurrences of 'llo': ${occurrences}`);
    console.log(`String time: ${elapsed}ms`);
}

main();
